src/components/protocol_handler/src/protocol_handler_impl.cc

```diff
@@ -2535,11 +2538,15 @@ void ProtocolHandlerImpl::GenerateServiceTransportsForStartSessionAck(
   LOG4CXX_AUTO_TRACE(logger_);
 
   if (service_transports.size() == 0) {
-    LOG4CXX_TRACE(logger_,
-                  "Empty Service Transports. Allowing service to run on both "
-                  "connections");
-    serviceTransports.push_back(1);
-    serviceTransports.push_back(2);
+    if (secondary_enabled) {
```

Comment:
`if (secondary_enabled && !secondary_transport_types.empty()) {` to cover the case where there aren't any secondary transports available in the first place

Reply:
Fixed in commit d4159c1768e28dccdd6027a714de59f93e06c8b2